lufopyramids.sh

```diff
@@ -16,81 +23,205 @@
+# 7. Upload tiles to blob storage
+#
 # Requirements:
 # -----------------------------------------
 # sudo apt-get install -y gdal-bin python-gdal
 #
 # -----------------------------------------
+# Usage:
+# -----------------------------------------
+# ./lufopyramids.sh <source dir of tiffs> <destination dir of result> <YYYY> <infrarood | luchtfotos>
+# ./lufopyramids.sh /mnt/lufo/2004 /mnt/lufo 2004 luchtfotos
+# OR
+# ./lufopyramids.sh /mnt/infrarood2018 /mnt/infrarood 2018 infrarood
+# -----------------------------------------
 #
-# In 2018/2019 the luchtfotos files need to be translated first to remove an alpha layer. Run this in the dir of the external harddrive
-# In the 2020 dataset this layer does not seem to exist anymore.
 #
-# for i in *.tif; do
-#     [ -f "$i" ] || break
-#     gdal_translate $i /mnt/lufo2020/$i -b 1 -b 2 -b 3 -mask 4 --config GDAL_TIFF_INTERNAL_MASK YES
-# done
+# When using ECW files, translate to GeoTiff manually:
+# https://hub.docker.com/r/ginetto/gdal
+# docker run --rm -it --name gdalecw -v /mnt/lufo/lufo2009/ams:/home/datafolder ginetto/gdal:2.4.4_ECW gdal_translate /home/datafolder/Amsterdam_groot.ecw /home/datafolder/lufo2009.tif
 #
-# cp -a *.aux /mnt/lufo2020
-# cp -a *.tfw /mnt/lufo2020
 
+### Set local vars
 SOURCEDIR=$1
 DESTDIR_ROOT=$2
 DESTDIR=$2/pyramid
 YEAR=$3
 TYPE=$4
 
-# Create a list of images in the source directory
+### Reset all tifs to use 3 bands:
+for i in $SOURCEDIR/*.tif; do
+    [ -f "$i" ] || break
+    gdal_translate $i $DESTDIR_ROOT/$(basename $i) -a_srs "EPSG:28992" -b 1 -b 2 -b 3
+done
+
+# ### Create a list of images in the source directory
+gdalbuildvrt -overwrite -resolution highest -r cubic -hidenodata -srcnodata "255 255 255" $DESTDIR_ROOT/overview.vrt $DESTDIR_ROOT/*.tif
 
-if [ -f /tmp/list.txt ] ; then
-    rm -r /tmp/list.txt
+### Clean up
+if [ -d $DESTDIR ] ; then
+   rm -rf $DESTDIR
 fi
 
-ls $SOURCEDIR/*.tif > /tmp/list.txt
+### Create pyramid base dir and retile to create pyramid with 5 levels in fixed tile sizes
+mkdir -p $DESTDIR/0
+gdal_retile.py -v -r cubic -resume -ps 8192 8192 -co TILED=YES -co COMPRESS=JPEG -co PHOTOMETRIC=YCBCR -s_srs "EPSG:28992" -targetDir $DESTDIR $DESTDIR_ROOT/overview.vrt
+# gdal_retile.py -v -r cubic -levels 5 -resume -ps 8192 8192 -co TILED=YES -co COMPRESS=JPEG -co PHOTOMETRIC=YCBCR -s_srs "EPSG:28992" -targetDir $DESTDIR $DESTDIR_ROOT/overview.vrt
 
-#if [ -d $DESTDIR ] ; then
-#    rm -rf $DESTDIR
-#fi
+### Put original resolution images in their own subdirectory
+mv $DESTDIR/*.tif $DESTDIR/0
 
-mkdir -p $DESTDIR/0
-gdal_retile.py -v -r cubic -levels 5 -ps 8192 8192 -co TILED=YES -co COMPRESS=JPEG -co PHOTOMETRIC=YCBCR -s_srs "EPSG:28992" -targetDir $DESTDIR --optfile /tmp/list.txt
+### Set Levels and loop over them
+### This exists to ensure public WMS usage is much quicker as the TIF files are in the blob storage
+LEVELS=$(find $DESTDIR/* -type d -exec basename {} \; | sort)
```

Comment:
As we only use the level 0 directory, no need for a while loop here

Reply:
We need the 5 levels in normal operations to provide quick WMS.
Level 0 is only used for seeding.